web/src/stream_popover.ts

```diff
@@ -445,9 +446,18 @@ export async function build_move_topic_to_stream_popover(
         // disabled in case when user does not have permission to edit
         // topic and thus submit button is disabled if stream is also
         // not changed.
-        util.the($<HTMLButtonElement>("#move_topic_modal button.dialog_submit_button")).disabled =
+        let is_disabled = false;
+        if (realm.realm_mandatory_topics && !new_topic_name?.trim()) {
```

Comment:
We use `new_topic_name?.trim() === ""` at other places, so we should do the same here as well.

Reply:
Is this same as [this](https://github.com/zulip/zulip/pull/32532#discussion_r1875450261) one?

Reply:
> We should also disable the button when the topic name is "(no topic)" like we do in validating while sending message.

Is this same as [this](https://github.com/zulip/zulip/pull/32532#discussion_r1875450261) one?

Reply:
I think you mean when new_topic_name === "(no topic)".

Reply:
This is not fixed yet to replace `!new_topic_name?.trim()` check as mentioned above. And yes button should also be disabled if `new_topic_name==="(no topic)"`.

Reply:
`!new_topic_name?.trim()` is still not replaced with `new_topic_name === ""` check. And also would be good to just have both `new_topic_name === ""` and `new_topic_name === "(no topic)"` check in the same if-block.